From bd77b1bd79e657a3c6252e1eb2439c421b11e0bc Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Wed, 27 Jan 2021 17:36:50 +0100
Subject: [PATCH] sudo: update to version 1.9.5p2

Fixes:

./sendlog.c: In function 'main':
./sendlog.c:1819:5: error: 'for' loop initial declarations are only allowed in C99 mode
     for (int i = 0; i < nr_of_conns; i++) {
     ^
./sendlog.c:1819:5: note: use option -std=c99 or -std=gnu99 to compile your code
---
 admin/sudo/Makefile | 30 ++++++++++++++++++------------
 1 file changed, 18 insertions(+), 12 deletions(-)

diff --git a/admin/sudo/Makefile b/admin/sudo/Makefile
index 43e92977d..e0293548c 100644
--- a/admin/sudo/Makefile
+++ b/admin/sudo/Makefile
@@ -8,14 +8,18 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=sudo
-PKG_VERSION:=1.8.14p3
+PKG_VERSION:=1.9.5p2
 PKG_RELEASE:=1
-PKG_LICENSE:=ISC
-PKG_LICENSE_FILES:=doc/LICENSE
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_URL:=http://www.sudo.ws/dist
-PKG_MD5SUM:=93dbd1e47c136179ff1b01494c1c0e75
+PKG_SOURCE_URL:=https://www.sudo.ws/dist
+PKG_HASH:=539e2ef43c8a55026697fb0474ab6a925a11206b5aa58710cb42a0e1c81f0978
+
+PKG_MAINTAINER:=Alexandru Ardelean <ardeleanalex@gmail.com>
+
+PKG_LICENSE:=ISC
+PKG_LICENSE_FILES:=doc/LICENSE
+PKG_CPE_ID:=cpe:/a:todd_miller:sudo
 
 PKG_INSTALL:=1
 
@@ -27,8 +31,9 @@ define Package/sudo
   SECTION:=admin
   CATEGORY:=Administration
   TITLE:=Delegate authority to run commands
-  URL:=http://www.sudo.ws/
+  URL:=https://www.sudo.ws/
   MAINTAINER:=Gergely Kiss <mail.gery@gmail.com>
+  DEPENDS:=+USE_UCLIBC:libpthread
 endef
 
 define Package/sudo/description
@@ -43,6 +48,8 @@ define Package/sudo/conffiles
 /etc/sudoers.d/
 endef
 
+TARGET_CFLAGS += --std=c99
+
 CONFIGURE_ARGS+= \
 	--without-pam \
 	--disable-pam-session \
@@ -71,18 +78,17 @@ endef
 $(eval $(call HostBuild))
 
 define Package/sudo/install
-	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_DIR) \
+		$(1)/etc/{init.d,sudoers.d} \
+		$(1)/usr/{bin,sbin} \
+		$(1)/usr/lib/sudo
+
 	$(CP) $(PKG_INSTALL_DIR)/usr/bin/sudo $(1)/usr/bin/
 	chmod 4755 $(1)/usr/bin/sudo
-	$(INSTALL_DIR) $(1)/usr/sbin
 	$(CP) $(PKG_INSTALL_DIR)/usr/sbin/visudo $(1)/usr/sbin/
-	$(INSTALL_DIR) $(1)/etc
 	$(CP) $(PKG_INSTALL_DIR)/etc/sudoers $(1)/etc/
 	chmod 0440 $(1)/etc/sudoers
-	$(INSTALL_DIR) $(1)/etc/sudoers.d
-	$(INSTALL_DIR) $(1)/usr/lib/sudo
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/sudo/*.so* $(1)/usr/lib/sudo/
-	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) ./files/sudo.init $(1)/etc/init.d/sudo
 endef
 
-- 
2.25.1

